crates/sats/src/db/def.rs

```diff
@@ -593,16 +632,8 @@ impl TableSchema {
         self.project(indexes.into_iter())
     }
 
-    pub fn get_row_type(&self) -> ProductType {
-        ProductType::new(
-            self.columns
-                .iter()
-                .map(|c| ProductTypeElement {
-                    name: Some(c.col_name.clone()),
-                    algebraic_type: c.col_type.clone(),
-                })
-                .collect(),
-        )
+    pub fn get_row_type(&self) -> &ProductType {
```

Comment:
Please leave a comment on this code explaining what it's purpose is and why we are caching the `product_type`